server/src/main/java/org/apache/druid/indexing/overlord/IndexerMetadataStorageCoordinator.java

```diff
@@ -124,6 +127,7 @@ default Collection<DataSegment> retrieveUsedSegmentsForInterval(
   Collection<DataSegment> retrieveUsedSegmentsForIntervals(
```

Comment:
Can these methods be used by third party extensions, which would fail compilation with the change in parameters. Should we instead add a new version of the function?

Reply:
They are only used by the overlord. The interface allows tests to have custom implementations as needed. Also, the new nullable `version` parameter shouldn't cause any issues with rolling upgrade and downgrade situations. @kfaraz, could you please spot check this assumption?

Reply:
This interface is not marked `@PublicApi` or otherwise exposed as an extension point. So it is safe to update methods as needed.

But, as @zachjsh suggests, I would prefer we add new methods here and anywhere else if applicable. We seem to be passing a lot of nulls right now from several places just because there is one usage of the method that needs to pass a non-null `versions`.

Reply:
👍 I can set the existing interface method as the default and invoke the new method inside it, so null is passed only from the default implementation.